gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,193 @@
+         return this._maximum;
+     }
+ 
+     public set maximum(value: number) {
+         if (this._maximum === value) {
+             return;
+         }
+ 
+         this._maximum = Math.max(value, this._minimum);
+         this.value = this._value;
+     }
+
+    /** Gets or sets current value */
+    public get value(): number {
+        return this._value;
+    }
+
+    public set value(value: number) {
+        value = Math.max(Math.min(value, this._maximum), this._minimum);
+
+        if (this._value === value) {
+            return;
+        }
+
+        this._value = value;
+        this.onValueChangedObservable.notifyObservers(value);
+    }
+
+    protected get start(): number{
+        if(!this.node){
+            return -1;
+        }
+
+        return this._sliderBar.position.x - this._sliderBar.scaling.y / 2;
+    }
+
+    protected get end(): number{
+        if(!this.node){
+            return 1;
+        }
+
+        return this._sliderBar.position.x + this._sliderBar.scaling.y / 2;
+    }
+
+    // Mesh association
+    protected _createNode(scene: Scene): TransformNode {
+        const anchor = new AbstractMesh(`${this.name}_slider`, scene);    
+        
+        this._sliderBar = MeshBuilder.CreateCylinder(`${this.name}_sliderbar`, { diameter: 0.03, height: 1.0 }, scene);
+        this._sliderBar.rotation.z = -Math.PI / 2;
+        this._sliderBar.scaling.y = 2;
+        this._sliderBar.isPickable = false;
+        this._sliderBar.setParent(anchor);
+
+        this._sliderThumb = MeshBuilder.CreateBox(`${this.name}_sliderthumb`, { size: 0.1 }, scene);
+        this._sliderThumb.scaling = new Vector3(2, 2, 2);
+        this._sliderThumb.position.x = this._convertToPosition(this.value);
+        this._sliderThumb.addBehavior(this._createBehavior());
+        this._sliderThumb.setParent(anchor);
+
+        return anchor;
+    }
+
+    protected _affectMaterial(mesh: AbstractMesh) {
+        this._sliderBarMaterial = this._createBarMaterial(mesh.getScene());
+        this._sliderBar.material = this._sliderBarMaterial;
+
+        this._sliderThumbMaterial = this._createThumbMaterial(mesh.getScene());
+        this._sliderThumb.material =  this._sliderThumbMaterial;
+    }
+    
+    private _createBarMaterial(scene: Scene): Material {
+        const material  =  new StandardMaterial(`${this.name}_sliderbar_material`, scene);
+        material.specularColor = Color3.Black();
+
+        return material;       
+    }
+
+    private _createThumbMaterial(scene: Scene): Material {
+        const material  =  new StandardMaterial(`${this.name}_sliderthumb_material`, scene);
+        material.specularColor = Color3.Black();
+
+        return material;       
+    }
+
+    private _createBehavior(): PointerDragBehavior {
+        const pointerDragBehavior = new PointerDragBehavior({ dragAxis: new Vector3(1, 0, 0) });
+        pointerDragBehavior.moveAttached = false;
+
+        pointerDragBehavior.onDragObservable.add((event) => {

```

Comment:
Is it possible to have the value be dynamically updated, instead of just on release? While keeping it set to specific values, of course. I've personally always gotten annoyed with scroll bars/other draggables that don't show me what I want until I release the button, meaning I can't browse options as easily.

Reply:
Agreed. May be we can add this as an option for the end user to pass when creating the slider. By default we can dynamically update the value onDrag. Thanks